### PR TITLE
ISPN-14143 + ISPN-14144 Task authorization 

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-14143
https://issues.redhat.com/browse/ISPN-14144

### CODE REVIEW

File: core/src/test/java/org/infinispan/security/ClusteredSecureCacheTest.java [hunk 1]
```diff
@@ -30,9 +51,16 @@ public CacheMode getCacheMode() {
...
       final ConfigurationBuilder builder = getDefaultClusteredCacheConfig(getCacheMode());
-      global.security().authorization().enable()
-            .principalRoleMapper(new IdentityRoleMapper()).role("admin").permission(AuthorizationPermission.ALL);
-      builder.security().authorization().enable().role("admin");
+      GlobalAuthorizationConfigurationBuilder globalRoles = global.security().authorization().enable()
+            .principalRoleMapper(new IdentityRoleMapper());
+      for (AuthorizationPermission perm : AuthorizationPermission.values()) {
+         globalRoles.role(perm.toString()).permission(perm);
+      }
+      global.serialization().addAdvancedExternalizer(4321, new SecureConsumer.Externalizer());
```
Comment (ryanemerson) [inline core/src/test/java/org/infinispan/security/ClusteredSecureCacheTest.java:59]: Why are we using an Externalizer here instead of a ProtoAdapter?

Comment (tristantarrant) [reply core/src/test/java/org/infinispan/security/ClusteredSecureCacheTest.java:59]: For the test, I thought it was quicker

**Comment by tristantarrant** [comment]
@ryanemerson failures are unrelated

**Comment by ryanemerson** [comment]
Thanks @tristantarrant

**Pull request merged**